src/platform/Tizen/AppPreference.cpp

```diff
@@ -19,145 +19,106 @@
-    }
-    else if (isExist == false)
-    {
-        err = CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND;
-    }
-
-    return err;
-}
-
 CHIP_ERROR CheckData(const char * key)
 {
-    return __IsKeyExist(key);
+    bool isExist = false;
+    int err      = preference_is_existing(key, &isExist);
+    VerifyOrReturnError(err == PREFERENCE_ERROR_NONE, CHIP_ERROR_INCORRECT_STATE);
+    return isExist ? CHIP_NO_ERROR : CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND;
 }
 
 CHIP_ERROR GetData(const char * key, void * data, size_t dataSize, size_t * getDataSize, size_t offset)
 {
-    CHIP_ERROR err                = CHIP_NO_ERROR;
-    int preErr                    = PREFERENCE_ERROR_NONE;
-    char * encodedData            = NULL;
-    uint8_t * decodedData         = NULL;
-    size_t encodedDataSize        = 0;
-    size_t encodedDataPaddingSize = 0;
-    size_t decodedDataSize        = 0;
-    size_t expectedDecodedSize    = 0;
-    size_t copy_size              = 0;
-
-    VerifyOrExit(data != NULL, err = CHIP_ERROR_INVALID_ARGUMENT);
-
-    err = __IsKeyExist(key);
-    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogProgress(DeviceLayer, "Not found data [%s]", key));
-
-    preErr = preference_get_string(key, &encodedData);
-    VerifyOrExit(preErr == PREFERENCE_ERROR_NONE, err = CHIP_ERROR_INCORRECT_STATE);
-    encodedDataSize = strlen(encodedData);
-
-    if ((encodedDataSize > 0) && (encodedData[encodedDataSize - 1] == '='))
-    {
-        encodedDataPaddingSize++;
-        if ((encodedDataSize > 1) && (encodedData[encodedDataSize - 2] == '='))
-            encodedDataPaddingSize++;
-    }
-    expectedDecodedSize = ((encodedDataSize - encodedDataPaddingSize) * 3) / 4;
-
-    decodedData = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(expectedDecodedSize));
-    VerifyOrExit(decodedData != NULL, err = CHIP_ERROR_INCORRECT_STATE);
+    char * encodedData = nullptr;
 
-    decodedDataSize = Base64Decode(encodedData, static_cast<uint16_t>(encodedDataSize), decodedData);
+    VerifyOrReturnError(data != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
 
-    copy_size = min(dataSize, decodedDataSize - offset);
-    if (getDataSize != NULL)
+    int err = preference_get_string(key, &encodedData);
+    if (err == PREFERENCE_ERROR_NO_KEY)
     {
-        *getDataSize = copy_size;
+        return CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND;
+    }
+    if (err == PREFERENCE_ERROR_OUT_OF_MEMORY)
+    {
+        return CHIP_ERROR_NO_MEMORY;
+    }
+    if (err != PREFERENCE_ERROR_NONE)
+    {
+        ChipLogError(DeviceLayer, "Failed to get preference [%s]: %s", key, get_error_message(err));
+        return CHIP_ERROR_INCORRECT_STATE;
     }
-    memset(data, 0, dataSize);
-    memcpy(data, decodedData + offset, copy_size);
 
-    ChipLogProgress(DeviceLayer, "Get data [%s:%s]", key, (char *) data);
+    // Make sure that string allocated by preference_get_string() will be freed
+    std::unique_ptr<char, decltype(&::free)> _{ encodedData, &::free };
```

Comment:
Can this line be put before/near the call to preference_get_string()? That will make it easier to understand that encodedData is alloc'd inside preference_get_string()

Reply:
Done